entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -255,48 +275,88 @@ class Neo4jRepository(
         return Pair(queryStatistics.nodesDeleted, queryStatistics.relationshipsDeleted)
     }
 
-    fun deleteEntityProperty(entityId: String, propertyName: String): Int {
+    fun deleteEntityProperty(subjectNodeInfo: SubjectNodeInfo, propertyName: String, datasetId: URI? = null, deleteAll: Boolean = false): Int {
         /**
          * Delete :
          *
          * 1. the property
          * 2. the properties of the property
          * 3. the relationships of the property
          */
-        val query = """
-            MATCH (entity:Entity { id: ${'$'}entityId })-[:HAS_VALUE]->(prop:Property { name: ${'$'}propertyName })
+
+        val query = if (deleteAll)
+            """
+            MATCH (entity:${subjectNodeInfo.label} { id: ${'$'}entityId })-[:HAS_VALUE]->(prop:Property { name: ${'$'}propertyName })
+            OPTIONAL MATCH (prop)-[:HAS_VALUE]->(propOfProp:Property)

```

Comment:
no way, there must be a way to avoid duplicating the below part everywhere, in both property & relationships method, maybe another method could fit..

```
OPTIONAL MATCH (prop)-[:HAS_VALUE]->(propOfProp:Property)
...
DETACH DELETE prop, propOfProp, relOfProp
```

Reply:
A method that returns this:
```
OPTIONAL MATCH (prop)-[:HAS_VALUE]->(propOfProp:Property)
...
DETACH DELETE prop, propOfProp, relOfProp
```
to be concatenated to the initial match ?